File: Week 6 - Python/sentimental-mario-more/mario.py
# Get the height of pyramind+
while True:
    try:
        height = int(input("Height: "))
    except ValueError:
        height = -1 
    if height >= 1 and height <= 8:
        break

# For each row of pyramind
for i in range(height):
    # For each column of pyramind
    for j in range((height + 1) * 2):
        # Print left spaces
        if j < height - i - 1:
            print(" ", end="")
        # Print left hashes
        elif j >= height - i - 1 and j < height:
            print("#", end="")
        # Print gap
        elif j == height or j == height + 1:
            print(" ", end="")
        # Print right hashes
        elif j > height + 1 and j < ((height + 1) * 2) - height + i + 1:
            print("#", end="")
    print()

File: Week 1 - C/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height = 0;

    // Get the height of pyramind
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // For each row of pyramind
    for (int i = 0; i < height; i++)
    {
        // For each column of pyramid
        for (int j = 0; j < (height + 1) * 2; j++)
        {
            // Print left spaces
            if (j < height - i - 1)
            {
                printf(" ");
            }
            // Print left hashes
            else if (j >= height - i - 1 && j < height)
            {
                printf("#");
            }
            // Print gap
            else if (j == height || j == height + 1)
            {
                printf(" ");
            }
            // Print right hashes
            else if (j > height + 1 && j < ((height + 1) * 2) - height + i + 1)
            {
                printf("#");
            }
        }

        printf("\n");
    }
}